packages/wasm-dpp/src/identity/state_transition/asset_lock_proof/chain/validation.rs

```diff
@@ -0,0 +1,46 @@
+use dpp::identity::state_transition::asset_lock_proof::{
+    AssetLockTransactionValidator, ChainAssetLockProofStructureValidator,
+};
+use dpp::state_transition::state_transition_execution_context::StateTransitionExecutionContext;
+use dpp::ProtocolError;
+use std::sync::Arc;
+use wasm_bindgen::prelude::*;
+
+use crate::buffer::Buffer;
+use crate::utils::ToSerdeJSONExt;
+use crate::{
+    errors::from_dpp_err,
+    state_repository::{ExternalStateRepositoryLike, ExternalStateRepositoryLikeWrapper},
+    validation::ValidationResultWasm,
+    StateTransitionExecutionContextWasm,
+};
+
+#[wasm_bindgen(js_name=validateChainAssetLockProofStructure)]
+pub async fn validate_chain_asset_lock_proof_structure(
```

Comment:
I'd probably made it into a structure, so you don't have to create the validation every time you run a validation

Reply:
Fixed